saleor/core/templatetags/taxed_prices.py

```diff
@@ -0,0 +1,43 @@
+from django import template
+from prices import MoneyRange, TaxedMoney, TaxedMoneyRange
+
+from ...core.utils.taxes import DEFAULT_TAX_RATE_NAME
+
+register = template.Library()
+
+
+@register.inclusion_tag('product/_price_range.html', takes_context=True)
+def price_range(context, price_range):
+    display_gross_prices = context['site'].settings.display_gross_prices
+    return {
+        'price_range': price_range,
+        'display_gross_prices': display_gross_prices}
+
+
+@register.simple_tag
+def tax_rate(taxes, rate_name):
+    """Return tax rate value for given tax rate type in current country."""
+    if not taxes:
+        return 0
+
+    tax = taxes.get(rate_name) or taxes.get(DEFAULT_TAX_RATE_NAME)
+    return tax['value']
+
+
+@register.inclusion_tag('price.html', takes_context=True)
+def price(context, base, display_gross=None, html=True):
+    if isinstance(base, (TaxedMoney, TaxedMoneyRange)):
```

Comment:
This check is done to seperate TaxedMoney prices conversion, that query for `display_gross_prices`, from usual flow.

Reply:
This just adds a little bit of  visual overhead to the function. If there's special flow for base being TaxedMoney or TaxedMoneyRange, I'd rather return early when base is not one of them. It will allow us to avoid multiple indentation 

eg.
```
@register.inclusion_tag('price.html', takes_context=True)
def price(context, base, display_gross=None, html=True):
    if not isinstance(base, (TaxedMoney, TaxedMoneyRange)):
        is_range = isinstance(base, MoneyRange)
        return {'price': base, 'is_range': is_range, 'html': html}

    if display_gross is None:
        display_gross = context['site'].settings.display_gross_prices

    if isinstance(base, TaxedMoneyRange):
        if display_gross:
            base = MoneyRange(start=base.start.gross, stop=base.stop.gross)
        else:
            base = MoneyRange(start=base.start.net, stop=base.stop.net)
    elif isinstance(base, TaxedMoney):
        base = base.gross if display_gross else base.net

    is_range = isinstance(base, MoneyRange)
    return {'price': base, 'is_range': is_range, 'html': html}
```
But that's a matter of styling preferences, leave it as you feel it 😄

Reply:
I second that, it's easier to read if the code checks for exceptions rather than for common cases.

Reply:
I would not say it is an exception, rather an additional conversion to perform. In the code snippet above we end up with two return statements doing the same, which, in my opinion, is harder to maintain.